### PR TITLE
Fixes incorrect end date timestamp calculation in order info.

### LINKED ISSUE #18: Incorrect end date time in order info
PR #14 fixes the checkout date in cart page, but still incorrect in order confirmation.

We'll be using [`DateTime::createFromFormat`](http://php.net/manual/en/datetime.createfromformat.php), which means we'll bump required version of PHP to >= 5.3.

### PR DESCRIPTION
**Pull request by gedex**
Resolves #18.

### CODE REVIEW

**Comment by gedex** [comment]
@dwainm Please review.

File: includes/class-wc-accommodation-booking-order-info.php [hunk 1]
```diff
@@ -53,19 +53,17 @@ public function add_checkinout_info_to_order_email( $item_id, $item, $order ) {
...
 	 * @param  int $duration
-	 * @return string
+	 * @return int
 	 */
 	private function get_end_date_timestamp( $start_date, $duration ) {
-		$format             = str_replace( array( 'Y','m','d' ), array( '%Y','%m','%d' ), get_option('date_format') );
-		$start_time_pieces  = strptime( $start_date, $format );
-		$end_date_timestamp = strtotime( ( $start_time_pieces['tm_mon'] + 1 ) . '/' . $start_time_pieces['tm_mday'] . '/' . ( $start_time_pieces['tm_year'] + 1900 ) );
-		$end_date           = strtotime( "+{$duration} day", $end_date_timestamp );
+		$datetime = DateTime::createFromFormat( get_option( 'date_format' ) . ' H:i:s', $start_date . ' 00:00:00' );
```
Comment (justinshreve) [inline includes/class-wc-accommodation-booking-order-info.php]: The reason I didn't originally do this is that it is only valid in PHP versions > 5.3.0 and we need to support 5.2.4+.

Comment (gedex) [inline includes/class-wc-accommodation-booking-order-info.php]: Yeah, I thought so. Unfortunately, characters format in `strptime` are different with what `date_format` uses  and might cause unpredictable bug in the future (for instance `F` has different meaning). We'll be adding PHP >= 5.3 as dependency (see #21) so it bails out if not satisfied.

**Comment by dwainm** [comment]
LG @gedex

**Pull request merged**